pkg/compact/compact.go

```diff
@@ -764,7 +766,11 @@ func (cg *Group) Compact(ctx context.Context, dir string, planner Planner, comp
 		return false, ulid.ULID{}, errors.Wrap(err, "create compaction group dir")
 	}
 
-	shouldRerun, compID, err := cg.compact(ctx, subDir, planner, comp)
+	var err error
+	tracing.DoInSpanWithErr(ctx, "group_compaction", func(ctx context.Context) error {
```

Comment:
All other spans are named `compaction_<x>`, should we make this follow the same convention?